### PR TITLE
Use `safe_extra` to standardize extra dependencies

### PR DESCRIPTION
**Pull request by deepyaman**
# Pull Request Check List

Resolves: #4123

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by deepyaman** [comment]
I can't quite figure out why the two running jobs are failing; the tests pass on my machine (aside from `venv`-related ones that didn't pass even when I first cloned the project), and the timeouts seem to happen before tests run.

**Review by finswimmer — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution @deepyaman :+1: 

I've talked about it with @sdispater and he's fine with being not case sensitiv here. But instead of using `pkg_resources.safe_extra` it would be better to implement our own helper method. Thus we can avoid a dependency that isn't absolutely necessary. The implementation seems to be trivial: https://github.com/pypa/pkg_resources/blob/6f81a44010d1266494025647dd1e1f0befa5b26b/pkg_resources/__init__.py#L1435

fin swimmer

**Comment by deepyaman** [comment]
> But instead of using `pkg_resources.safe_extra` it would be better to implement our own helper method.

@finswimmer Sorry for the delay, but this is done!

**Review by neersighted — COMMENTED** [review]
This looks good to me -- can we rebase this so it can be merged?

**Comment by deepyaman** [comment]
> Looks like we still need to get CI working -- also there's a lot of junk commits. Could we squash this PR down to 1-3 (logical/meaningful) commits?

Sorry, I had to step out, but I was in the middle of setting up my local env to resolve the issues. It's been a year, so a lot has changed, including my laptop. :)

I'll ping you once I've got them resolved, and I can also squash down to a change or two (I wrongly assumed or forgot that Poetry doesn't just squash-merge everything).

**Comment by deepyaman** [comment]
@neersighted Think we're good now! I'll just update with rebase since there's been more pushes, but I don't imagine there will be any issues.

**Comment by neersighted** [comment]
> > Looks like we still need to get CI working -- also there's a lot of junk commits. Could we squash this PR down to 1-3 (logical/meaningful) commits?
> 
> As an unrelated note on this, looking at the (recent) commit history, there's a mix of Conventional Commit and old-school sentence-case commits. Maybe it makes sense to standardize on one?

We're not picky yet but eventually I'll get around to gating commits on conventional commit compliance :smile:

And we do squash merge when needed, but during a more involved change (with a developer that knows how to rebase) like this I prefer to have logical commits to make bisecting easier -- e.g. in this case I'd prefer add tests then add code to make the change.

**Comment by deepyaman** [comment]
> I'd prefer add tests then add code to make the change.

@neersighted Do you want me to do this, or is this good to go as is? I've never broken commits down like that (because generally try to ensure that tests pass on each commit on the main branch on projects I've worked on), but I can update it if that's the standard for Poetry and what we need to close this out.

**Comment by neersighted** [comment]
> > I'd prefer add tests then add code to make the change.
> 
> @neersighted Do you want me to do this, or is this good to go as is? I've never broken commits down like that (because generally try to ensure that tests pass on each commit on the main branch on projects I've worked on), but I can update it if that's the standard for Poetry and what we need to close this out.

It's nothing enforced -- merely a preference for review, and given we're already reviewed this change one commit is fine :smile:

File: src/poetry/utils/helpers.py [hunk 1]
```diff
@@ -157,3 +157,15 @@ def pluralize(count: int, word: str = "") -> str:
     if count == 1:
         return word
     return word + "s"
+
+
+def safe_extra(extra: str) -> str:
+    """Convert an arbitrary string to a standard 'extra' name
```
Comment (neersighted) [inline src/poetry/utils/helpers.py:163]:
```suggestion
    """Convert an arbitrary string to a standard 'extra' name.
```

**Review by neersighted — APPROVED** [review]

**Comment by deepyaman** [comment]
@neersighted Thank you so much for your help in closing this out!

**Pull request merged**